## Reanna Laurell

### Hypothesis
If I have had instagram since 2013 then my number of likes should be around 1,000 in total.

The statistical part of my hypothesis comes from me assuming that in last 10 years I have atleast liked 100 posts a year. This is also suppose to cover for the days that I wasn't on Instagram due to traveling as well as for recently when I haven't been on it has much. The theoretical part of my hypothesis is assuming that 1,000 post is a big enough number to cover having Instagram for the last 10 years.

The data included in liked_post.json file is appropiate for testing my hypthesis because it will be able to show the total number of likes I had within the ten year span, as well as more information on each post that I liked, including the time stamp.

Instagram collected and stored the data of my liked over the years that I have had the app. They collected this data and kept it to better understand what type of content that I liked more and found reels and ads to connect with those interests. This data is reliable in displaying information on users posts that I have liked over the years as well as the time frame of when I liked the post. This data may be unreliable as it goes further back in time as Instagram might not have kept detailed reports from then or it could be unreliable because I downloaded this information roughly two weeks ago and have liked posts since then as well. 

### Data

In [1]:
import pandas as pd
import json

Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.
Intel MKL WARNING: Support of Intel(R) Streaming SIMD Extensions 4.2 (Intel(R) SSE4.2) enabled only processors has been deprecated. Intel oneAPI Math Kernel Library 2025.0 will require Intel(R) Advanced Vector Extensions (Intel(R) AVX) instructions.


In [2]:
with open("/Users/reannalaurell/Library/Mobile Documents/com~apple~CloudDocs/Downloads/instagram-reanna.r.21 2/likes/liked_posts.json") as j: 
    my_likes = json.load(j)

In [3]:
my_likes

{'likes_media_likes': [{'title': 'byspencerthomas',
   'string_list_data': [{'href': 'https://www.instagram.com/reel/CukjwMJA6Kr/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693417607}]},
  {'title': 'abigail_van_drunen',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwknjE-O-7v/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693417528}]},
  {'title': 'msu_football',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwjTszsrTEC/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693393708}]},
  {'title': 'kuna.2',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwjcWdwMmfX/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693393696}]},
  {'title': 'thomasrhettakins',
   'string_list_data': [{'href': 'https://www.instagram.com/p/CwjG2pYMKWM/',
     'value': 'ð\x9f\x91\x8d',
     'timestamp': 1693393596}]},
  {'title': 'gomezselena_22',
   'string_list_data': [{'href': 'https://www.instagram.com/reel/CtWbHkshlIW/',
     'value': 

In [4]:
likes_data = pd.DataFrame(my_likes['likes_media_likes'])

In [120]:
likes_data.tail()

title                                   string_list_data  \
3988   canisiuswsoccer  [{'href': 'https://www.instagram.com/p/Ch7UWu2...   
3989        annaxsitar  [{'href': 'https://www.instagram.com/p/Ch3mOxC...   
3990              jess  [{'href': 'https://www.instagram.com/p/Ch66l2w...   
3991   haleyhightowerr  [{'href': 'https://www.instagram.com/p/Ch5uCVg...   
3992  libby_zacharyasz  [{'href': 'https://www.instagram.com/p/Ch5nSzO...   

       timestamp  
3988  1661985161  
3989  1661962282  
3990  1661943724  
3991  1661914519  
3992  1661898981

In [12]:
type(my_likes['likes_media_likes'])

list

In [25]:
my_likes['likes_media_likes'][256]['string_list_data'][0]['timestamp']

1691630134

In [28]:
times = [x['string_list_data'][0]['timestamp'] for x in my_likes['likes_media_likes']]

In [122]:
times

[1693417607,
 1693417528,
 1693393708,
 1693393696,
 1693393596,
 1693359959,
 1693353541,
 1693353309,
 1693349385,
 1693348199,
 1693335353,
 1693328816,
 1693328114,
 1693327881,
 1693317207,
 1693262116,
 1693262053,
 1693262044,
 1693251410,
 1693247847,
 1693242971,
 1693236699,
 1693196914,
 1693196351,
 1693196286,
 1693186928,
 1693186802,
 1693185806,
 1693165875,
 1693165785,
 1693161169,
 1693083746,
 1693082435,
 1693082364,
 1693065396,
 1693059905,
 1693059664,
 1693007474,
 1693004553,
 1692997812,
 1692968131,
 1692968091,
 1692925958,
 1692925932,
 1692919549,
 1692919538,
 1692888384,
 1692836489,
 1692830049,
 1692813042,
 1692812883,
 1692812876,
 1692812842,
 1692812834,
 1692800377,
 1692800167,
 1692790254,
 1692755908,
 1692755670,
 1692755494,
 1692755445,
 1692753968,
 1692739469,
 1692715665,
 1692715598,
 1692685115,
 1692666892,
 1692654908,
 1692654780,
 1692641266,
 1692635445,
 1692635374,
 1692635284,
 1692635184,
 1692625400,
 1692625365,
 1692625135,

In [30]:
likes_data['timestamp'] = times

In [119]:
likes_data.head()

title                                   string_list_data  \
0     byspencerthomas  [{'href': 'https://www.instagram.com/reel/Cukj...   
1  abigail_van_drunen  [{'href': 'https://www.instagram.com/p/CwknjE-...   
2        msu_football  [{'href': 'https://www.instagram.com/p/CwjTszs...   
3              kuna.2  [{'href': 'https://www.instagram.com/p/CwjcWdw...   
4    thomasrhettakins  [{'href': 'https://www.instagram.com/p/CwjG2pY...   

    timestamp  
0  1693417607  
1  1693417528  
2  1693393708  
3  1693393696  
4  1693393596

In [111]:
likes_data.groupby('title').count().head(20).sort_values('timestamp', ascending = False)

string_list_data  timestamp
title                                                     
4paws_crick_                                  4          4
_amysophia                                    3          3
__tbaker99__                                  3          3
_alex_perez1                                  2          2
4paws_lavender                                2          2
1.direction_ot5                               1          1
917vis                                        1          1
_akshujain_                                   1          1
__wherewewander                               1          1
__thelittlesunflower.reads___                 1          1
_.shroyer._                                   1          1
8eyestheband                                  1          1
1000libraries                                 1          1
3amaashima                                    1          1
392.dr3                                       1          1
21.79am                                       1          1
20sandthriving                                1          1
1k_dean                                       1          1
1989.mp3                                      1          1
_ashley_michaels_                             1          1

In [118]:
likes_data.groupby('title').count().sort_values('timestamp', ascending = False).head()

string_list_data  timestamp
title                                         
handmade.by.regan                55         55
msu_football                     50         50
uswnt                            40         40
jess                             38         38
annaxsitar                       33         33

### What I learned from the data

Once I had created the pandas Data Frame, I was able to gain more information on my likes over the years as well. One thing that was surprising is shown by the line, likes_data.groupby('title').count().sort_values('timestamp', ascending = False).head(). From this chart I was able to see that the most amount of likes to one account have come from recent interactions with these accounts. Handmade.by.regan, is my sister greeting card account that she has only had for about six months at most now. Going down from there the account I have probably been following the longest from the last five is USWNT and that doesn't even have the most likes overall. 

While looking at the front of the pandas Data Frame based on sorting in alphbetical order, showed that the most recent likes have come from a 4 Paws account that I follow. I find this very fitting as it is impossible to not like a dog image when it appears on your screen. 

### Conclusion

Based on what I found from the likes_data method, my total number of likes is actually around 3988, with this being stated, it shows that my hypothesis was off by 2,988. This was surpising to me how off I was as it didn't occur to me before now how much I had used Instagram thoughout the years. From this however, I have been able to learn how to go about making a hypothesis, import the data into a jupyter notebook, and manipulate the data into different Data Frames and sorting methods to further test and find answers from my data.  